app/scripts/lib/nonce-tracker.js

```diff
@@ -83,6 +76,43 @@ class NonceTracker {
     return mutex
   }
 
+  async _getNetworkNonceAndDetails (address) {
+    // calculate next nonce
+    // we need to make sure our base count
+    // and pending count are from the same block
+    const currentBlock = await this._getCurrentBlock()
+    const blockNumber = currentBlock.blockNumber
+    const pendingTransactions = this.getPendingTransactions(address)
+    const pendingCount = pendingTransactions.length
+    assert(Number.isInteger(pendingCount), `nonce-tracker - pendingCount is not an integer - got: (${typeof pendingCount}) "${pendingCount}"`)
+    const baseCountHex = await this._getTxCount(address, currentBlock)
+    const baseCount = parseInt(baseCountHex, 16)
+    assert(Number.isInteger(baseCount), `nonce-tracker - baseCount is not an integer - got: (${typeof baseCount}) "${baseCount}"`)
+    const networkNonce = baseCount + pendingCount
```

Comment:
I don't think `pendingCount` is the right metric to use, since multiple pending txs can have the same nonce.

Once this state is reached, this code would always suggest a too-high nonce, which would add to the wrongness of future nonce calculations.